central/processlisteningonport/datastore/datastore_impl_test.go

```diff
@@ -318,84 +265,100 @@ func (suite *PLOPDataStoreTestSuite) TestPLOPAddOpenTwice() {
 
 	// Add PLOP referencing those indicators
 	suite.NoError(suite.datastore.AddProcessListeningOnPort(
-		suite.hasWriteCtx, fixtureconsts.Cluster1, plopObjects...))
-
-	// Add the same PLOP again
-	suite.NoError(suite.datastore.AddProcessListeningOnPort(
-		suite.hasWriteCtx, fixtureconsts.Cluster1, plopObjects...))
-
-	// Fetch inserted PLOP back
-	newPlops, err := suite.datastore.GetProcessListeningOnPort(
-		suite.hasWriteCtx, fixtureconsts.Deployment1)
-	suite.NoError(err)
+		suite.hasAllCtx, fixtureconsts.Cluster1, plopObjects...))
 
-	suite.Len(newPlops, 1)
-	suite.Equal(newPlops[0], &storage.ProcessListeningOnPort{
-		ContainerName: "test_container1",
-		PodId:         fixtureconsts.PodName1,
-		PodUid:        fixtureconsts.PodUID1,
-		DeploymentId:  fixtureconsts.Deployment1,
-		ClusterId:     fixtureconsts.Cluster1,
-		Namespace:     fixtureconsts.Namespace1,
-		Endpoint: &storage.ProcessListeningOnPort_Endpoint{
-			Port:     1234,
-			Protocol: storage.L4Protocol_L4_PROTOCOL_TCP,
+	cases := map[string]struct {
+		ctx           context.Context
+		expectAllowed bool
+	}{
+		"all access": {
+			ctx:           sac.WithGlobalAccessScopeChecker(context.Background(), sac.AllowAllAccessScopeChecker()),
+			expectAllowed: true,
 		},
-		Signal: &storage.ProcessSignal{
-			Name:         "test_process1",
-			Args:         "test_arguments1",
-			ExecFilePath: "test_path1",
+		"access to cluster and namespace": {
+			ctx: sac.WithGlobalAccessScopeChecker(context.Background(), sac.AllowFixedScopes(
+				sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+				sac.ResourceScopeKeys(resources.Namespace),
+				sac.ClusterScopeKeys(fixtureconsts.Cluster1),
+				sac.NamespaceScopeKeys(fixtureconsts.Namespace1),
+			)),
+			expectAllowed: true,
+		},
+		"read and write access": {
```

Comment:
This test fails. I don't understand why.

Reply:
Try passing `storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS` in the context `AccessModeScopeKeys`, the drill down for read access might actually look for the READ key rather than consider the READ_WRITE one contains the read one.

Reply:
Done. It works now.